AsyncDisplayKit/ASImageNode.mm

```diff
@@ -111,7 +111,9 @@ - (instancetype)initWithViewBlock:(ASDisplayNodeViewBlock)viewBlock
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDN::MutexLocker l(_imageLock);
-  if (_image)
+  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO)
+    return [super calculateSizeThatFits:constrainedSize];
```

Comment:
Could you add a comment here?  I'm still not totally sure why this works.  You're not passing in preferredFrameSize as the argument to the super method, so I'm not sure where the actual value of preferredFrameSize is used in the ASImageNode case.  Is an alternative to just change the default behavior to lay out at the constrainedSize if there is no image set yet?